ci/cd view: Build status is always completed and green even when it build fails

<!--- Provide a general summary of the issue in the Title above -->

## Expected Behavior
- Status indicator on the each build step is correct colour and status text

## Current Behavior
- When the build fails each build step is listed as `Completed` with a green status mark

## Possible Solution
- Update the status text and colour to reflect a more realistic status

## Steps to Reproduce
- Have a component with builds enabled like the CRA template
- Create a failing build
- Head to the CI/CD tab, and check out the failing build steps

Hi @benjdlambert 

Fixed the `Failed to fetch` errors, thanks to your `yarn start-backend` tip.

Tried creating components several times using `Create... > Create React App Template` on the website. They took a long time and would eventually fail.

Installed Docker. Tried running `yarn run docker-build:app`, `yarn run docker-build`, `docker run --rm -it -p 7000:7000 -e NODE_ENV=development example-backend:latest`, as well as `docker-compose up`. 

They failed at
```
Backend failed to start up Error: Cannot find module '../schema'
Require stack:
- /usr/src/app/plugins/catalog-graphql/dist/index.cjs.js
- /usr/src/app/plugins/graphql/dist/index.cjs.js
- /usr/src/app/packages/backend/dist/index.cjs.js
...
``` 

Fortunately, trying `Create... > Create React App Template` after the Docker installation helped complete a template build named `testing`.
However, even after creating the template, I could not find the `testing` template on the website. 

Any pointers to help me find the built CRA template and creating the failing build (similar to the screenshots posted by @shmidt-i) would do wonders.

Also tried `yarn add @backstage/plugin-github-actions` and `yarn workspace @backstage/plugin-github-actions start` using https://backstage.io/docs/plugins/ but I'm probably not doing it right.

I tried looking into the docs but I am having trouble fully understanding the docs. 
Really want to get this done, but I'm running into several beginner roadblocks.

Thank you so much for being patient.

--- a/plugins/github-actions/src/components/Cards/Cards.tsx
+++ b/plugins/github-actions/src/components/Cards/Cards.tsx
@@ -60,7 +60,10 @@ const WidgetContent = ({
       metadata={{
         status: (
           <>
-            <WorkflowRunStatus status={lastRun.status} />
+            <WorkflowRunStatus
+              status={lastRun.status}
+              conclusion={lastRun.conclusion}
+            />
           </>
         ),
         message: lastRun.message,

--- a/plugins/github-actions/src/components/WorkflowRunDetails/WorkflowRunDetails.tsx
+++ b/plugins/github-actions/src/components/WorkflowRunDetails/WorkflowRunDetails.tsx
@@ -106,7 +106,10 @@ const StepView = ({ step }: { step: Step }) => {
         />
       </TableCell>
       <TableCell>
-        <WorkflowRunStatus status={step.status.toUpperCase()} />
+        <WorkflowRunStatus
+          status={step.status.toUpperCase()}
+          conclusion={step.conclusion.toUpperCase()}
+        />
       </TableCell>
     </TableRow>
   );
@@ -204,7 +207,10 @@ export const WorkflowRunDetails = ({ entity }: { entity: Entity }) => {
                 <Typography noWrap>Status</Typography>
               </TableCell>
               <TableCell>
-                <WorkflowRunStatus status={details.value?.status} />
+                <WorkflowRunStatus
+                  status={details.value?.status}
+                  conclusion={details.value?.conclusion}
+                />
               </TableCell>
             </TableRow>
             <TableRow>

--- a/plugins/github-actions/src/components/WorkflowRunStatus/WorkflowRunStatus.tsx
+++ b/plugins/github-actions/src/components/WorkflowRunStatus/WorkflowRunStatus.tsx
@@ -14,13 +14,22 @@
  * limitations under the License.
  */
 
-import { StatusPending, StatusRunning, StatusOK } from '@backstage/core';
+import {
+  StatusPending,
+  StatusRunning,
+  StatusOK,
+  StatusWarning,
+  StatusAborted,
+  StatusError,
+} from '@backstage/core';
 import React from 'react';
 
 export const WorkflowRunStatus = ({
   status,
+  conclusion,
 }: {
   status: string | undefined;
+  conclusion: string | undefined;
 }) => {
   if (status === undefined) return null;
   switch (status.toLowerCase()) {
@@ -37,11 +46,32 @@ export const WorkflowRunStatus = ({
         </>
       );
     case 'completed':
-      return (
-        <>
-          <StatusOK /> Completed
-        </>
-      );
+      switch (conclusion?.toLowerCase()) {
+        case 'skipped' || 'canceled':
+          return (
+            <>
+              <StatusAborted /> Aborted
+            </>
+          );
+        case 'timed_out':
+          return (
+            <>
+              <StatusWarning /> Timed out
+            </>
+          );
+        case 'failure':
+          return (
+            <>
+              <StatusError /> Error
+            </>
+          );
+        default:
+          return (
+            <>
+              <StatusOK /> Completed
+            </>
+          );
+      }
     default:
       return (
         <>

--- a/plugins/github-actions/src/components/WorkflowRunsTable/WorkflowRunsTable.tsx
+++ b/plugins/github-actions/src/components/WorkflowRunsTable/WorkflowRunsTable.tsx
@@ -46,6 +46,7 @@ export type WorkflowRun = {
     };
   };
   status: string;
+  conclusion: string;
   onReRunClick: () => void;
 };
 
@@ -84,7 +85,7 @@ const generatedColumns: TableColumn[] = [
 
     render: (row: Partial<WorkflowRun>) => (
       <Box display="flex" alignItems="center">
-        <WorkflowRunStatus status={row.status} />
+        <WorkflowRunStatus status={row.status} conclusion={row.conclusion} />
       </Box>
     ),
   },

--- a/plugins/github-actions/src/components/useWorkflowRuns.ts
+++ b/plugins/github-actions/src/components/useWorkflowRuns.ts
@@ -87,6 +87,7 @@ export function useWorkflowRuns({
                 },
               },
               status: run.status,
+              conclusion: run.conclusion,
               url: run.url,
               githubUrl: run.html_url,
             }));

